src/utils/hooks/useRestartApplication.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  ApplicationPatch,
+  TemplateElementCategory,
+  Trigger,
+  useRestartApplicationMutation,
+} from '../generated/graphql'
+import { FullStructure } from '../types'
+
+// below lines are used to get return type of the function that is returned by useRestartApplicationMutation
+type UseRestartApplicationMutation = ReturnType<typeof useRestartApplicationMutation>
+type promiseReturnType = ReturnType<UseRestartApplicationMutation[0]>
+// hook used to submit review, , as per type definition below (returns promise that resolve with mutation result data)
+type UseRestartApplication = (serial: string) => (structure: FullStructure) => promiseReturnType
+
+type ConstructRestartApplicationPatch = (structure: FullStructure) => ApplicationPatch
+
+const useRestartApplication: UseRestartApplication = (serial) => {
+  const [restartApplicationMutation] = useRestartApplicationMutation()
+
+  const constructRestartApplicationPatch: ConstructRestartApplicationPatch = (structure) => {
+    const reviewableElements = Object.values(structure.elementsById || {}).filter(
```

Comment:
Maybe is it worth to add a comment saying we allow all responses to be reviewed - not only changes requested?

Reply:
8707cac86e2254a5f2b43674252d946bbe0f77c6